File: app/src/main/java/ru/geekbrains/notes/note/DateEditSorterComparator.java
package ru.geekbrains.notes.note;

import java.util.Comparator;

public class DateEditSorterComparator implements Comparator<Note> {
    @Override
    public int compare(Note note1, Note note2) {
        return Long.compare(note1.getDateEdit(), note2.getDateEdit());
    }
}

File: app/src/main/java/ru/geekbrains/notes/note/comparator/DateEditSorterComparator.kt
package ru.geekbrains.notes.note.comparator

import ru.geekbrains.notes.note.Note
import java.util.*

class DateEditSorterComparator : Comparator<Note> {
    override fun compare(note1: Note, note2: Note): Int {
        return java.lang.Long.compare(note1.dateEdit, note2.dateEdit)
    }
}